### ISSUE TITLE
Opening  "compatible" text files in the emacs session running the query

### ISSUE THREAD

**Issue by oatmealm** [juergenhoetzel/consult-gnome-search#1]
Hi there! Thanks for writing this, very useful. I re-enabled tracker in my Fedora40 just to explore further :)

Sorry for the awkward title...

I've noticed that the Files results seem to open via whatever xdg mechanism and open in the default app, which is emacs.desktop. Which means it'll open the file in a new frame, but also, when I do it from my non default instance of Emacs (using --init-directory ~/not-emacs.d or ~/.config/emacs etc.) , it'll launch the emacs.dekstop with the default profile in parallel...

Is there a way to get the file path retrieved from the search provider and simply open it in the active session running consult-gnome-search (if the file type is supported, i.e. org and txt files etc.)?

**Comment by oatmealm** [comment]
Just to add that other wise, from emacs launched form default emacs.desktop it'll open a new frame as far as I can tell, which also not desirable I guess.

**Comment by juergenhoetzel** [comment]
Thank you for your interest in consult-gnome-search

> I've noticed that the Files results seem to open via whatever xdg mechanism and open in the default app, which is emacs.desktop. Which means it'll open the file in a new frame, but also, when I do it from my non default instance of Emacs (using --init-directory ~/not-emacs.d or ~/.config/emacs etc.) , it'll launch the emacs.dekstop with the default profile in parallel...

Have you tried setting the default app to `emacsclient`. This should prevent opening the file it in a new Emacs instance:

```bash
xdg-mime default emacsclient.desktop text/plain
xdg-mime default emacsclient.desktop text/x-csrc
```
> 
> Is there a way to get the file path retrieved from the search provider and simply open it in the active session running consult-gnome-search (if the file type is supported, i.e. org and txt files etc.)?

I thought about implementing this myself in order to avoid the “detour” via the search provider and xdg-mime mechanism.

In fact, `Nautilus`  returns the file path via the ID. However, this seems to me to be an implementation detail of Nautilus and I am not sure if it is a good idea to parse this ID from Emacs. I could make it optional (via a defcustom).

```elisp
(dbus-call-method :session "org.gnome.Nautilus" "/org/gnome/Nautilus/SearchProvider" "org.gnome.Shell.SearchProvider2"  "GetInitialResultSet" '("weight.org"))
("file:///home/juergen/shared/org/weight.org" "file:///home/juergen/shared/org/weight.org_archive")
```

**Comment by oatmealm** [comment]
Thanks. I have the mime types sey, but it still launch a separate instance for me, even though a server is active in a running instance of emacs.  I'll investigate further  (must be something specific to my setup).

**Comment by juergenhoetzel** [comment]
This should be fixed by #2

The default value for `consult-gnome-search-activate-functions` already provides a function to directly activate the Nautilus search result within Emacs.
